BootCampV2/Homeworks/ThirdWeek/Services/UserFileManager.cs

```diff
@@ -0,0 +1,258 @@
+using BootCampV2.Homeworks.ThirdWeek.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace BootCampV2.Homeworks.ThirdWeek
+{
+    public enum LoginResult
+    {
+        AccountWithUsernameDontExists,
+        InvalidPassword,
+        Success
+    }
+
+    public enum RegistrationResult
+    {
+        InvalidUsername,
+        NotMatchingPassword,
+        Success
+    }
+
+    public class UserFileManager
+    {
+        public readonly string Path;
+
+        private List<Account> Accounts = new List<Account>();
+
+        public UserFileManager(string path)
+        {
+            Path = path;
+
+            bool validFile = ValidateUserFile(Path, out List<Account> accounts, out Exception exception);
+
+            if (validFile)
+                Accounts = accounts;
+            else
+                throw exception;
+        }
+
+        public LoginResult Login(string username, string password, out Account loggedInAccount)
+        {
+            loggedInAccount = null;
+
+            Account account = GetAccountByUsername(username);
+
+            if (account == null)
+                return LoginResult.AccountWithUsernameDontExists;
+            else if (!password.Equals(account.Password))
+                return LoginResult.InvalidPassword;
+            else
+                loggedInAccount = account;
```

Comment:
Add brackets even for one lines or inline them. This way of writing is very prone to error.
```
{
}
```